### PR TITLE
Add wasm foundation

### PR DESCRIPTION
**Pull request by KendallWeihe**

### CODE REVIEW

File: bound/kt/pom.xml
Comment (KendallWeihe) [inline bound/kt/pom.xml:11]: todo: undo this prior to merge

File: LICENSE
Comment (KendallWeihe) [inline LICENSE:1]: todo: fix this before merging (accidental)

File: Justfile
Comment (KendallWeihe) [inline Justfile:11]: @diehuxx I know this is no bueno but thinking of leaving it for now. I had to do it because of the `wasm-pack` version we're using... though we could look into moving to a lower `wasm-pack` I'm not sure what the consequences would be. Lmk your thoughts.

File: bindings/web5_wasm/src/crypto/key_managers.rs
Comment (KendallWeihe) [inline bindings/web5_wasm/src/crypto/key_managers.rs:48]: here's an example of a _foreign_ implementation, for which we'll use this same pattern for a foreign `fetch` implementation

File: crates/web5/src/http.rs
Comment (KendallWeihe) [inline crates/web5/src/http.rs:1]: @diehuxx hold off on reviewing anything http-related until we have #354 merged, and I'll wrangle this around to use `http-std` as the default and enable developer-override

File: crates/web5/src/jose.rs
Comment (KendallWeihe) [inline crates/web5/src/jose.rs:29]: oh btw @diehuxx @nitro-neal forgot to mention earlier, but I've also added `Jws` here b/c tbdex uses a detatched JWS for message signatures

File: Justfile
Comment (diehuxx) [reply Justfile:11]: That's fine. We CI job that checks if the web5 crate has MSRV 1.74. The CI job checks `crates/web5/Cargo.toml` to get `rust-version = 1.74`, so as long as we don't change that, we're fine.

The only drawback of using a higher rust version during development is that we may accidentally use features in `crates/web5` that aren't supported in version 1.74. If that happens, we'll just notice them in the CI job failure and fix afterward.

File: bound/typescript/src/crypto/key_managers/index.ts
Comment (nitro-neal) [inline bound/typescript/src/crypto/key_managers/index.ts:23]: its strange throwing a function, 

Chad says we can do this:
```
// Define the decorator
function CatchWeb5Error(
  target: Object,
  propertyKey: string,
  descriptor: PropertyDescriptor
) {
  const originalMethod = descriptor.value;

  descriptor.value = function (...args: any[]) {
    try {
      return originalMethod.apply(this, args);
    } catch (error) {
      throw catchWeb5Error(error);
    }
  };

  return descriptor;
}

// Apply the decorator to your methods
class YourClass {
  @CatchWeb5Error
  computeThumbprint(): string {
    const wasmJwk = this.toWasmJwk();
    return wasmJwk.compute_thumbprint();
  }

  @CatchWeb5Error
  anotherMethod(): void {
    // Your method implementation
  }
}
```

Comment (nitro-neal) [reply bound/typescript/src/crypto/key_managers/index.ts:23]: We can think about it and if we want to go this way or another can make an issue to update error handling across the board

Comment (KendallWeihe) [reply bound/typescript/src/crypto/key_managers/index.ts:23]: But yeah, in a follow up PR

File: crates/web5/src/credentials/sign.rs
Comment (nitro-neal) [inline crates/web5/src/credentials/sign.rs:41]: nice

**Review by nitro-neal — APPROVED** [review]

File: bound/typescript/src/crypto/key_managers/index.ts
Comment (KendallWeihe) [reply bound/typescript/src/crypto/key_managers/index.ts:23]: looks like decorators are still considered an experimental feature so I may settle on a higher order function for now

File: crates/web5/src/lib.rs
Comment (diehuxx) [inline crates/web5/src/lib.rs:7]: didn't catch this before merge, but why are we exposing `jose` now? IIRC we wanted to keep this as an implementation detail.

Comment (KendallWeihe) [reply crates/web5/src/lib.rs:7]: tbdex uses detached compact jws as it's signature field

**Pull request merged**
